src/bp-core/bp-core-options.php

```diff
@@ -1573,6 +1573,27 @@ function bp_register_confirm_email( $default = false ) {
 	return (bool) apply_filters( 'bp_register_confirm_email', (bool) bp_get_option( 'register-confirm-email', $default ) );
 }
 
+/**
+ * Display legal agreement field in registrations.
+ *
+ * @since BuddyBoss 1.5.8.4
+ *
+ * @param bool $default Optional. Fallback value if not found in the database.
+ *                      Default: false.
+ * @return bool True if Whether or not display legal agreement field in registrations.
+ */
+function bp_register_legal_agreement( $default = false ) {
+
+	/**
+	 * Filters whether or not display legal agreemnet field in registrations.
+	 *
+	 * @since BuddyBoss 1.5.8.4
+	 *
+	 * @param bool $value whether or not display legal agreemnet field in registrations.
+	 */
+	return (bool) apply_filters( 'bp_register_legal_agreement', (bool) bp_get_option( 'register-legal-agreement', $default ) );
+}
```

Comment:
```suggestion
function bb_register_legal_agreement( $default = false ) {

	/**
	 * Filters whether or not display legal agreement field in registrations.
	 *
	 * @since BuddyBoss 1.5.8.4
	 *
	 * @param bool $value whether or not display legal agreement field in registrations.
	 */
	return (bool) apply_filters( 'bb_register_legal_agreement', (bool) bp_get_option( 'register-legal-agreement', $default ) );
}
```